src/internal_events/datadog_metrics.rs

```diff
@@ -13,17 +12,18 @@ impl InternalEvent for DatadogMetricsEncodingError {
     fn emit_logs(&self) {
         error!(
             message = "Failed to encode Datadog metrics.",
-            error_type = error_type::ENCODER_FAILED,
             error = %self.error,
+            error_code = "failed_encoding_datadog_metrics",
+            error_type = error_type::ENCODER_FAILED,
```

Comment:
It looks like there can actually be a number of errors that occur here:

https://github.com/vectordotdev/vector/blob/f3bae2eea33f7d490954f6d35fb1b8c044eecf47/src/sinks/datadog/metrics/request_builder.rs#L13-L32

Can we differentiate these errors with their own codes?

I'd also recommend using more generic codes that can be shared across events like `encoding_failed` rather than `failed_encoding_datadog_metrics`.

Reply:
Good point, will update that 😉